### ISSUE TITLE
Deployments

### ISSUE THREAD

**Issue by srikanthankam** [kubernetes/website#34099]
Hello Everyone,

I have an application pod created using Kubernetes deployment, most of the times within seconds application pods are going to maximum number (10) and for information min. replicas mentioned is 1.

Question is there any way I can make delay between creating replicas?

Please suggest any solution.

**Comment by howieyuen** [comment]
when a Deployment is created, kube-controller-manager will create a ReplicaSet to do version control (upgrade or roll-back), and ReplicaSet controller batch create Pod immediately since all Pods are equal to other, so there is no need to make delay for that.
I don't know what do your want for the delay, may be you need a StatefulSet which can guarantee one-by-one created in order?

**Comment by srikanthankam** [comment]
Actually, my application runs migration scripts and also came to know that at a time all replicas should not run those scripts.

In addition, I tried with Statefulset but the problem I experienced is replicas are following stickiness means picking old image instead of latest.

So, what I am thinking is if there is an option to delay the replicas, it would help app to run scripts one after the other.
